runtime/parser/parse_metrics_view.go

```diff
@@ -582,13 +644,25 @@ func (p *Parser) parseMetricsView(node *Node) error {
 		}
 	}
 
+	// validate and insert inline explore, if true and no error is returned from the method then an explore resource is created so no error should be returned after this point
+	skipExplore, exploreRes, err := p.parseInlineExplore(tmp, node.Name, node.Paths)
+	if err != nil {
+		return fmt.Errorf("failed to parse inline explore: %w", err)
+	}
+
+	// insert metrics view resource immediately after parsing the inline explore as it inserts the explore resource so we should not return an error now
 	r, err := p.insertResource(ResourceKindMetricsView, node.Name, node.Paths, node.Refs...)
 	if err != nil {
+		// If we fail to insert the metrics view, we must delete the inline explore if it was created.
+		if exploreRes != nil {
+			p.deleteResource(exploreRes)
+		}
```

Comment:
Calling `deleteResource` is not safe, it should only be called from the main parser loop. 

It needs to return the error if a dry run was not done earlier. If a dry run was already done (if explore was not skipped), I think it should panic because it's in an inconsistent state then.

Reply:
Forgot to change this one, already doing dry insert for mv